src/DotNetLightning.Core/Crypto/Keys.fs

```diff
@@ -239,6 +239,27 @@ type ChannelPrivKeys = {
             HtlcBasepoint = this.HtlcBasepointSecret.HtlcBasepoint()
         }
 
+    member this.SignWithFundingPrivKey (psbt: PSBT)
+                                           : TransactionSignature * PSBT =
+        let fundingPubKey = this.FundingPrivKey.FundingPubKey()
+        psbt.SignWithKeys(this.FundingPrivKey.RawKey()) |> ignore
+        match psbt.GetMatchingSig(fundingPubKey.RawPubKey()) with
+        | Some signature -> (signature, psbt)
+        | None -> failwithf "Failed to get signature for %A with funding pub key (%A). This should never happen" psbt fundingPubKey
+
+    member this.SignHtlcTx (psbt: PSBT)
+                           (perCommitmentPoint: PerCommitmentPoint)
+                               : TransactionSignature * PSBT =
+        let htlcPrivKey = perCommitmentPoint.DeriveHtlcPrivKey this.HtlcBasepointSecret
+        let htlcPubKey = htlcPrivKey.HtlcPubKey()
+        psbt.SignWithKeys(htlcPrivKey.RawKey()) |> ignore
+        match psbt.GetMatchingSig(htlcPubKey.RawPubKey()) with
+        | Some signature -> (signature, psbt)
+        | None ->
+            failwithf
+                "failed to get htlc signature for %A. with htlc pubkey (%A) and perCommitmentPoint (%A)"
+                psbt htlcPubKey perCommitmentPoint
+
```

Comment:
it seems this hunk was not meant to go in this commit

Reply:
I've rewritten the commit message to more accurately describe what the commit is doing, including mentioning that it moves these methods across to the `ChannelPrivKeys` type.